src/client/http_client.rs

```diff
@@ -332,17 +335,17 @@ impl HTTPClient {
                 }
             };
 
-            Ok(HttpResponse::new(content, status, seed))
+            Ok(HttpStreamResponse::new(stream, status, seed, rt))
         })
     }
 
-    pub fn analytics_query_request(
+    pub fn analytics_query_stream_request(
         &self,
         request: AnalyticsQueryRequest,
         deadline: Instant,
         ctrl_c: Arc<AtomicBool>,
-    ) -> Result<HttpResponse, ClientError> {
-        let rt = Runtime::new().unwrap();
+        rt: Arc<Runtime>,
+    ) -> Result<(Pin<Box<ResultsStream>>, u16), ClientError> {
```

Comment:
This is a bit confusing to me, why doesn't this return HttpStreamResponse?

Reply:
That was my mistake, corrected now